pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -1393,7 +1393,37 @@ def _check_unnecessary_comprehension(self, node):
         else:
             return
         if expr_list == target_list != []:
-            self.add_message("unnecessary-comprehension", node=node)
+            args = None
+            if isinstance(node.parent, astroid.DictComp) and isinstance(
+                utils.safe_infer(node.iter), astroid.objects.DictItems
```

Comment:
You use `utils.safe_infer` multiple times here. It would be better to do it once and than use the result:
```py
inferred = utils.safe_infer(node.iter)
if ...
```

Reply:
Fixed: 0912d67